### PR TITLE
Update SA score to use FingerprintGenerators

### PR DESCRIPTION
**Pull request by UnixJunkie**
#### Reference Issue
Fixes https://github.com/rdkit/rdkit/issues/7794

#### What does this implement/fix? Explain your changes.
The SA score is impacted by the new fingerprint interface.

#### Any other comments?
I tested on 1000 lead-like molecules that the new scores are the same as the old scores.

### CODE REVIEW

**Comment by UnixJunkie** [comment]
Maybe, someone lucky enough to attend the rdkit UGM 2024 will be able to push the merge button for this PR.
Thanks!

PS: have fun, lucky you. :-)

File: Contrib/SA_Score/sascorer.py [hunk 1]
```diff
@@ -46,14 +46,15 @@ def numBridgeheadsAndSpiro(mol, ri=None):
   nBridgehead = rdMolDescriptors.CalcNumBridgeheadAtoms(mol)
   return nBridgehead, nSpiro
 
+# 4294967295 was the default length of the Morgan fingerprint
+mfpgen = rdFingerprintGenerator.GetMorganGenerator(radius=2, fpSize=4294967295)
```
Comment (greglandrum) [inline Contrib/SA_Score/sascorer.py:50]: You should be using `GetSparseCountFingerprint()` below, in which case you do not need to specify the size for the FP here:
```suggestion
mfpgen = rdFingerprintGenerator.GetMorganGenerator(radius=2)
```

File: Contrib/SA_Score/sascorer.py [hunk 2]
```diff
@@ -46,14 +46,15 @@ def numBridgeheadsAndSpiro(mol, ri=None):
...
 
 def calculateScore(m):
   if _fscores is None:
     readFragmentScores()
 
   # fragment score
-  fp = rdMolDescriptors.GetMorganFingerprint(m,
-                                             2)  # <- 2 is the *radius* of the circular fingerprint
+  fp = mfpgen.GetCountFingerprint(m)
```
Comment (greglandrum) [inline Contrib/SA_Score/sascorer.py:57]:
```suggestion
  fp = mfpgen.GetSparseCountFingerprint(m)
```

**Review by greglandrum — CHANGES_REQUESTED** [review]
Suggested changes to use the fingerprint generator as they are intended to be used

**Comment by UnixJunkie** [comment]
The proposed changes have been committed, and my regression test is still OK (results unchanged compared to old version). So, I think this PR is ready to be merged.

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
Thanks @UnixJunkie !

**Pull request merged**